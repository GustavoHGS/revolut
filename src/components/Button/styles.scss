@import "assets/_variables.scss";

.button {
  border-radius: 5px;
  min-width: 80px;
  min-height: 30px;
  color: $white;
  font-family: GloberSemiBold;
  border: none;
  padding: 0px 8 0px 8;
  box-shadow: 0 2px 10px 2px rgba(69, 65, 78, 0.08);
  transition: all 0.4s ease-in-out 0s;
}

.button:hover {
  cursor: pointer;
}

.primary {
  background-color: $primary;
  outline: none;
}

.primary:hover {
  background-color: $blue-dark;
}

.primary:focus {
  border: 2px solid $blue-dark;
}


.secondary {
  background-color: $secondary;
}

.disabled {
  cursor: not-allowed;
  background-color: $gray-300;
}

.disabled:hover {
  cursor: not-allowed;
  background-color: $gray-300;
}

.disabled:focus {
  border: none;
}