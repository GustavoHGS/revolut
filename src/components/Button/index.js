/* eslint react/forbid-prop-types: off */
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import './styles.scss'

export default class Button extends Component {
  state = {}

  handleClick = () => {
    const { onClick } = this.props
    onClick()
  }

  render() {
    const { label, buttonType, secondary, className, disabled } = this.props
    return (
      <button
        onClick={this.handleClick}
        type={buttonType}
        className={
          classNames('button',
            secondary ? 'secondary' : 'primary',
            className,
            disabled ? 'disabled' : '')
        }
      >
        {label}
      </button>
    )
  }
}

Button.propTypes = {
  onClick: PropTypes.func,
  label: PropTypes.string,
  buttonType: PropTypes.string,
  secondary: PropTypes.bool,
  className: PropTypes.string,
  disabled: PropTypes.bool,
}

Button.defaultProps = {
  onClick: () => {},
  label: '',
  buttonType: 'button',
  secondary: false,
  className: '',
  disabled: false,
}
