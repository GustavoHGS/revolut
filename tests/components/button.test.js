import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Button from '../../src/components/Button'

describe('Button Component', () => {
  it ('matches the button snapshot', () => {
    const nodeTree = renderer.create(<Button />).toJSON()
    expect(nodeTree).toMatchSnapshot()
  })

  it('render props correctly', () => {
    const wrapper = shallow(<Button label="test" />)
    expect(wrapper.props().children).toEqual('test')
    expect(wrapper.find('button').contains('test')).toEqual(true)
  })
  it('has default button classname', () => {
    const wrapper = shallow(<Button label="test" />)
    expect(wrapper.hasClass('button')).toEqual(true)
  })
})
