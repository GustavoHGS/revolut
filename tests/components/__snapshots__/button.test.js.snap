// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component matches the button snapshot 1`] = `
<button
  className="button primary"
  onClick={[Function]}
  type="button"
>
  
</button>
`;
